roles/default/kiali-deploy/tasks/openshift/os-main.yml

```diff
@@ -26,6 +26,8 @@
     - "templates/openshift/role-controlplane.yaml"
     - "templates/openshift/rolebinding.yaml"
     - "templates/openshift/rolebinding-controlplane.yaml"
+    - "templates/openshift/clusterrole-oauth.yaml"
+    - "templates/openshift/clusterrolebinding-oauth.yaml"
```

Comment:
Only create the oauth cluster role stuff if we need them (that is, if auth strategy is `openshift`):
```suggestion
    - "{{ 'templates/openshift/clusterrole-oauth.yaml' if kiali_vars.auth.strategy == 'openshift' else '' }}"
    - "{{ 'templates/openshift/clusterrolebinding-oauth.yaml' if kiali_vars.auth.strategy == 'openshift' else '' }}"
```
Here's how to test this:

1. In kiali/kiali repo, `make/Makefile.operator.mk` set operator-cluster-role-creator to false (we never made this configurable via env var - probably should do that one day :)). This change:
```diff
diff --git a/make/Makefile.operator.mk b/make/Makefile.operator.mk
index 317f0251d..450881220 100644
--- a/make/Makefile.operator.mk
+++ b/make/Makefile.operator.mk
@@ -83,7 +83,7 @@ operator-create: .ensure-operator-repo-exists .ensure-operator-helm-chart-exists
     --helm-set                      "image.pullSecrets={${OPERATOR_IMAGE_PULL_SECRET_NAME}}" \
-    --operator-cluster-role-creator "true" \
+    --operator-cluster-role-creator "false" \
     --operator-image-name           "${CLUSTER_OPERATOR_INTERNAL_NAME}" \
```
2. Push and deploy the operator to the cluster - this operator will not have permission to create cluster roles: `make cluster-push operator-create`
3. Now try to install Kiali server with auth strategy of openshift - this should fail with permission errors: `make -e AUTH_STRATEGY="openshift" -e ACCESSIBLE_NAMESPACES="istio-system" kiali-create`
4. You can confirm step 3 fails by watching the progress in the Kiali CR - you will eventually see an error logged in the status field: `oc get kiali kiali -n kiali-operator -oyaml -w`
5. Now update that Kiali CR so that the auth strategy is anonymous: `make -e AUTH_STRATEGY="anonymous" -e ACCESSIBLE_NAMESPACES="istio-system" kiali-create`
6. Eventually, Kiali should be installed. Wait for it in istio-system -- `oc get pods -n istio-system -w`
7. Navigate to the Kiali route and see that you get to the UI without logging in, confirming anonymous mode was set.
8. Notice there are no cluster roles/bindings created for the server; you should only have the clusterroles for the operator itself:
```
 $ oc get clusterroles,clusterrolebindings -o name | grep kiali
clusterrole.rbac.authorization.k8s.io/kiali-operator
clusterrolebinding.rbac.authorization.k8s.io/kiali-operator
```